### PR TITLE
first cut of tcp/ip module

### PR DESCRIPTION
**Pull request by pski**

### CODE REVIEW

File: src/esp/main/io.c [hunk 1]
```diff
@@ -10,18 +15,75 @@
...
+	    switch( protocol ) {
+		    case PROTOCOL_PREFIX_TCPIP:
+				while (tcp_z80_out(read_byte()) != IP_STATE_SEND_TO_Z80);
+				tcp_get_send_buffer(&buf, &size);
+			    break;
+		    default:
+			    if ( rs_z80_out(protocol) != RS_STATE_SEND )
+			    	while (rs_z80_out(read_byte()) != RS_STATE_SEND);
+				rs_get_send_buffer(&buf, &size);
+			    break;
```
Comment (apuder) [inline src/esp/main/io.c]: Inconsistent indentation.

Comment (pski) [reply src/esp/main/io.c]: Completely fair comment about tabs/spaces.  I agree.  I paid zero attention to it.  Will pretty it up.

**Comment by pski** [comment]
I just noticed I need to clean up some of the other files.  Let me do that first.

**Comment by pski** [comment]
Ok, more clean up committed.
I will look into doxygen.

**Comment by apuder** [comment]
Thanks for the PR!

**Pull request merged**